src/udisks2monitor.cpp

```diff
@@ -651,6 +671,46 @@ void UDisks2::Monitor::doFormat(const QString &devicePath, const QString &dbusOb
     });
 }
 
+void UDisks2::Monitor::doSetLabel(const QString &devicePath, const QString &dbusObjectPath, const QVariantList &arguments)
+{
+    const QString dbusMethod = UDISKS2_BLOCK_SETLABEL;
+    if (devicePath.isEmpty()) {
+        qCCritical(lcMemoryCardLog) << "Cannot" << dbusMethod.toLower() << "without device name";
+        return;
+    }
+
+    QDBusMessage method = QDBusMessage::createMethodCall(
+                UDISKS2_SERVICE,
+                dbusObjectPath,
+                UDISKS2_FILESYSTEM_INTERFACE,
+                dbusMethod);
+    method.setArguments(arguments);
+    QDBusPendingCall pendingCall = QDBusConnection::systemBus().asyncCall(method);
+    QDBusPendingCallWatcher *watcher = new QDBusPendingCallWatcher(pendingCall, this);
+    connect(watcher, &QDBusPendingCallWatcher::finished,
+            this, [this, devicePath, dbusMethod](QDBusPendingCallWatcher *watcher) {
+        if (watcher->isValid() && watcher->isFinished()) {
+            emit status(devicePath, Partition::Unmounted);
```

Comment:
On error this stays forever on Renaming?

Reply:
For small polish could move the status changes after the 'if', on both branches it's the last thing done.

Reply:
Done.